### ISSUE TITLE
Feature: Add "Disconnect" option to app's Wi-Fi selector

### ISSUE THREAD

**Issue by mcous** [Opentrons/opentrons#4849]
## overview

As a user with a robot connected to Wi-Fi, I want to disconnect that robot from Wi-Fi because I've don't want other people on the network to be able to connect to the robot.

This ticket is for tracking the addition of a "Disconnect" option to the Wi-Fi select UI that exercises the disconnect endpoint added by #4847. It takes over part of the UI described in #2718

## acceptance criteria

- [ ] User presented an option to "Disconnect from ${network}" in Wi-Fi select field
    - https://zpl.io/anY8n9Y
- [ ] After clicking that option, the user is given a confirmation modal prior to actually disconnecting
    - https://zpl.io/bA43g4O
- [ ] Clicking the currently connected network in the selector does nothing

## implementation notes

- Blocked by API endpoint addition in #4847
- Blocked by `react-select` unification refactor in #4844

#2718 describes functionality that disables the Wi-Fi selector when not currently connected to the robot via Ethernet. This functionality is **out of scope** for this ticket and will be addressed by a separate ticket
